[hipsparse] build fails

The following are problems with hipsparse PKGBUILD:

- `"hip>=$pkgver"` dependency provides an error.

- `make DESTDIR="$pkgdir" install` needs to be used instead of `make install`

- line `cp -r "$srcdir/build/opt" "$pkgdir/"` refers to a directory that doesn't exist

In addition, the build depends on `rocsparse-config.cmake`. In that file the line `find_dependency(hip)` causes a search for Findhip.cmake which fails because at least according to the AUR packages the file is called FindHIP.cmake. Build works if the file instead says `find_dependency(HIP)`

Even after

```
$ sudo cp /opt/rocm/hip/cmake/FindHIP.cmake  /opt/rocm/hip/cmake/Findhip.cmake
```

and changing to HIP in `cmake/Dependencies.cmake`.

I still have issues building rocsparse from this repo. The aur version is out of date and does install fine, but does not seem to be compatible with the hipsparse from this repo.

Any idea what this is complaining about?

I assume its a dependency that is out of date in the AUR as well, but I don't really want to install everything manually. I'll be back to this after dinner.

```
$ makepkg
==> Making package: rocsparse 3.3.0-2 (Mo 01 Jun 2020 17:22:35 CEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found rocsparse-3.3.0.tar.gz
==> Validating source files with sha256sums...
    rocsparse-3.3.0.tar.gz ... Passed
==> Extracting sources...
  -> Extracting rocsparse-3.3.0.tar.gz with bsdtar
==> Removing existing $pkgdir/ directory...
==> Starting build()...
-- HCC compiler set; ROCm backend selected [ CXX=/opt/rocm/hcc/bin/hcc cmake .. ]
CMake Warning (dev) at /usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:272 (message):
  The package name passed to `find_package_handle_standard_args` (PkgConfig)
  does not match the name of the calling package (hcc).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  /usr/share/cmake-3.17/Modules/FindPkgConfig.cmake:45 (find_package_handle_standard_args)
  /opt/rocm/hcc/lib/cmake/hcc/ImportedTargets.cmake:1 (include)
  /opt/rocm/hcc/lib/cmake/hcc/hcc-config.cmake:54 (include)
  cmake/Dependencies.cmake:38 (find_package)
  CMakeLists.txt:85 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:272 (message):
  The package name passed to `find_package_handle_standard_args` (PkgConfig)
  does not match the name of the calling package (hcc).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  /usr/share/cmake-3.17/Modules/FindPkgConfig.cmake:45 (find_package_handle_standard_args)
  /opt/rocm/hcc/lib/cmake/hcc/ImportedTargets.cmake:1 (include)
  /opt/rocm/hcc/lib/cmake/hcc/hcc-config.cmake:54 (include)
  /usr/share/cmake-3.17/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm/hip/lib/cmake/hip/hip-config.cmake:77 (find_dependency)
  cmake/Dependencies.cmake:39 (find_package)
  CMakeLists.txt:85 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring done
-- Generating done
-- Build files have been written to: /home/pepper-jk/code/rocm-arch/rocsparse/src/build
[  2%] Building CXX object library/CMakeFiles/rocsparse.dir/src/extra/rocsparse_csrgemm.cpp.o
Call parameter type does not match function signature!
  %StackGuardSlot = alloca i8*, addrspace(5)
 i8**  call void @llvm.stackprotector(i8* %0, i8* addrspace(5)* %StackGuardSlot)
in function _Z39csrgemm_fill_block_per_row_host_pointerI21rocsparse_complex_numIdELj1024ELj32ELj2048ELj137EEviPKiS3_T_S3_S3_PKS4_S3_S3_S6_S4_S3_S3_S6_S3_PiPS4_21rocsparse_index_base_S9_S9_S9_bb
LLVM ERROR: Broken function found, compilation aborted!
Generating AMD GCN kernel failed in llc for target: gfx908
Call parameter type does not match function signature!
  %StackGuardSlot = alloca i8*, addrspace(5)
 i8**  call void @llvm.stackprotector(i8* %0, i8* addrspace(5)* %StackGuardSlot)
in function _Z39csrgemm_fill_block_per_row_host_pointerI21rocsparse_complex_numIdELj1024ELj32ELj2048ELj137EEviPKiS3_T_S3_S3_PKS4_S3_S3_S6_S4_S3_S3_S6_S3_PiPS4_21rocsparse_index_base_S9_S9_S9_bb
LLVM ERROR: Broken function found, compilation aborted!
Generating AMD GCN kernel failed in llc for target: gfx906
Call parameter type does not match function signature!
  %StackGuardSlot = alloca i8*, addrspace(5)
 i8**  call void @llvm.stackprotector(i8* %0, i8* addrspace(5)* %StackGuardSlot)
in function _Z39csrgemm_fill_block_per_row_host_pointerI21rocsparse_complex_numIdELj1024ELj32ELj2048ELj137EEviPKiS3_T_S3_S3_PKS4_S3_S3_S6_S4_S3_S3_S6_S3_PiPS4_21rocsparse_index_base_S9_S9_S9_bb
LLVM ERROR: Broken function found, compilation aborted!
Generating AMD GCN kernel failed in llc for target: gfx900
Call parameter type does not match function signature!
  %StackGuardSlot = alloca i8*, addrspace(5)
 i8**  call void @llvm.stackprotector(i8* %0, i8* addrspace(5)* %StackGuardSlot)
in function _Z39csrgemm_fill_block_per_row_host_pointerI21rocsparse_complex_numIdELj1024ELj32ELj2048ELj137EEviPKiS3_T_S3_S3_PKS4_S3_S3_S6_S4_S3_S3_S6_S3_PiPS4_21rocsparse_index_base_S9_S9_S9_bb
LLVM ERROR: Broken function found, compilation aborted!
Generating AMD GCN kernel failed in llc for target: gfx803
Error: hc-kernel-assemble[166]: failed with status -1
clang-10: error: HC assembler command failed with exit code 255 (use -v to see invocation)
make[2]: *** [library/CMakeFiles/rocsparse.dir/build.make:304: library/CMakeFiles/rocsparse.dir/src/extra/rocsparse_csrgemm.cpp.o] Error 255
make[1]: *** [CMakeFiles/Makefile2:114: library/CMakeFiles/rocsparse.dir/all] Error 2
make: *** [Makefile:172: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
```

Well if I try to build hipsparse (with rocsparse installed from AUR), I get this:

```
$ makepkg
==> Making package: hipsparse 3.3.0-4 (Mo 01 Jun 2020 18:35:33 CEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found hipsparse-3.3.0.tar.gz
==> Validating source files with sha256sums...
    hipsparse-3.3.0.tar.gz ... Passed
==> Extracting sources...
  -> Extracting hipsparse-3.3.0.tar.gz with bsdtar
==> Removing existing $pkgdir/ directory...
==> Starting build()...
CMake Warning (dev) at /usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:272 (message):
  The package name passed to `find_package_handle_standard_args` (HIP) does
  not match the name of the calling package (hip).  This can lead to problems
  in calling code that expects `find_package` result variables (e.g.,
  `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  /opt/rocm/hip/cmake/Findhip.cmake:175 (find_package_handle_standard_args)
  /usr/share/cmake-3.17/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm/rocsparse/lib/cmake/rocsparse/rocsparse-config.cmake:89 (find_dependency)
  cmake/Dependencies.cmake:34 (find_package)
  CMakeLists.txt:66 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at cmake/Dependencies.cmake:34 (find_package):
  Found package configuration file:

    /opt/rocm/rocsparse/lib/cmake/rocsparse/rocsparse-config.cmake

  but it set rocsparse_FOUND to FALSE so package "rocsparse" is considered to
  be NOT FOUND.  Reason given by package:

  rocsparse could not be found because dependency hip could not be found.

Call Stack (most recent call first):
  CMakeLists.txt:66 (include)


-- Configuring incomplete, errors occurred!
See also "/home/pepper-jk/code/rocm-arch/hipsparse/src/build/CMakeFiles/CMakeOutput.log".
==> ERROR: A failure occurred in build().
    Aborting...
```

And if I just change change `find_dependency(hip)` to `find_dependency(HIP)` in `/opt/rocm/rocsparse/lib/cmake/rocsparse/rocsparse-config.cmake` I still get:

<details><summary>output</summary>
<p>

```
$ makepkg                                                            
==> Making package: hipsparse 3.3.0-4 (Mo 01 Jun 2020 18:39:14 CEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found hipsparse-3.3.0.tar.gz
==> Validating source files with sha256sums...
    hipsparse-3.3.0.tar.gz ... Passed
==> Extracting sources...
  -> Extracting hipsparse-3.3.0.tar.gz with bsdtar
==> Removing existing $pkgdir/ directory...
==> Starting build()...
-- Configuring done
-- Generating done
-- Build files have been written to: /home/pepper-jk/code/rocm-arch/hipsparse/src/build
Scanning dependencies of target hipsparse
[ 50%] Building CXX object library/CMakeFiles/hipsparse.dir/src/hcc_detail/hipsparse.cpp.o
In file included from /home/pepper-jk/code/rocm-arch/hipsparse/src/hipSPARSE-rocm-3.3.0/library/src/hcc_detail/hipsparse.cpp:24:
/home/pepper-jk/code/rocm-arch/hipsparse/src/hipSPARSE-rocm-3.3.0/library/include/hipsparse.h:41:10: fatal error: 'hip/hip_complex.h' file not found
#include <hip/hip_complex.h>
         ^~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [library/CMakeFiles/hipsparse.dir/build.make:83: library/CMakeFiles/hipsparse.dir/src/hcc_detail/hipsparse.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:114: library/CMakeFiles/hipsparse.dir/all] Error 2
make: *** [Makefile:172: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
```

</p>
</details>

--- a/hipsparse/.SRCINFO
+++ b/hipsparse/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = hipsparse
 	pkgdesc = rocSPARSE marshalling library.
 	pkgver = 3.5.0
-	pkgrel = 2
+	pkgrel = 3
 	url = https://rocmdocs.amd.com/en/latest/ROCm_Libraries/ROCm_Libraries.html#hipsparse
 	arch = x86_64
 	license = MIT

--- a/hipsparse/PKGBUILD
+++ b/hipsparse/PKGBUILD
@@ -3,7 +3,7 @@
 
 pkgname=hipsparse
 pkgver=3.5.0
-pkgrel=2
+pkgrel=3
 pkgdesc='rocSPARSE marshalling library.'
 arch=('x86_64')
 url='https://rocmdocs.amd.com/en/latest/ROCm_Libraries/ROCm_Libraries.html#hipsparse'
@@ -28,7 +28,7 @@ build() {
   CXX=/opt/rocm/hip/bin/hipcc \
   cmake -B build -Wno-dev \
         -DCMAKE_CXX_STANDARD=20 \
-        -DCMAKE_INSTALL_PREFIX=opt/rocm \
+        -DCMAKE_INSTALL_PREFIX=/opt/rocm \
         -Damd_comgr_DIR=/opt/rocm/lib/cmake/amd_comgr \
         -DAMDDeviceLibs_DIR=/opt/rocm/lib/cmake/AMDDeviceLibs \
         -Dhip_DIR=/opt/rocm/hip/lib/cmake/hip \

